spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/TwoStepsConversions.java

```diff
@@ -205,14 +286,35 @@ public Value convertOnWrite(Object proppertyVal, DatastorePersistentProperty per
 		return writeConverter.apply(val);
 	}
 
-	private EntityValue convertOnWriteSingleEmbedded(Object val, String kindName) {
+	private EntityValue applyEntityValueBuilder(String kindName,
```

Comment:
Refactor this method to accept kind name and return builder. Refactor methods that use this one appropriately.

Reply:
No, this would cause functions that use this function to repeat `EntityValue.of(builder.build())`

Reply:
Exactly, that is much more readable!

Reply:
I respectfully disagree :P